json_test.go

```diff
@@ -23,9 +23,9 @@ func TestJSONUnmarshalAuthenticationMD5Password(t *testing.T) {
 }
 
 func TestJSONUnmarshalAuthenticationSASL(t *testing.T) {
-	data := []byte(`{"Type":"AuthenticationSASL", "AuthMechanisms":[]}`)
+	data := []byte(`{"Type":"AuthenticationSASL","AuthMechanisms":["SCRAM-SHA-256"]}`)
```

Comment:
Would only having this test without the Unmarshal implementation code already be enough?

Reply:
Yes, it works without the custom implementation, good catch.